### ISSUE TITLE
Dynamic Styling when uploading new image

### ISSUE THREAD

**Issue by alexvco** [thoughtbot/paperclip#2443]
I have a polymorphic association: Profile has one Image, Blog has many images.
I tried to follow along your code in the README but that does not work in this case.

models/blog.rb
class Blog < ApplicationRecord
  has_many :images, as: :imageable, dependent: :destroy
  accepts_nested_attributes_for :images, allow_destroy: true
end

models/profile.rb
class Profile < ApplicationRecord
  has_one :image, as: :imageable, dependent: :destroy
  accepts_nested_attributes_for :image, allow_destroy: true
end

models/image.rb
class Image < ActiveRecord::Base
belongs_to :imageable, :polymorphic => true

  Paperclip.interpolates :imageable_type do |attachment, style|
    attachment.instance.imageable_type.downcase
  end

  Paperclip.interpolates :imageable_id do |attachment, style|
    attachment.instance.imageable_id
  end

has_attached_file :attachment, 
styles: lambda { |attachment| {
      square: (attachment.instance.imageable_type.eql?("Profile") ? ["300x300#", 'jpg'] : {}),
      medium: (attachment.instance.imageable_type.eql?("Blog") ? ["300x190>", 'jpg'] : {}),
      large: (attachment.instance.imageable_type.eql?("Blog") ? ["825x523>", 'jpg'] : {})
    }
  }

Basically, if the imageable type is Profile, only upload the square style to S3 (do not bother uploading the medium and large styles), however currently it is uploading all styles to s3 with the original file size...

Your help is always appreciated

**Comment by gregogalante** [comment]
Get the same problem with dynamic styles.

Current code:

```ruby
# UserAttachment.
class UserAttachment < ApplicationRecord

  has_attached_file :file,
                    styles: ->(a) { a.instance.paperclip_styles },
                    path: ':rails_root/public/system/user_attachments/:id/:style_:filename',
                    url: '/system/user_attachments/:id/:style_:filename'

  validates_attachment_presence :file
  validates_attachment_content_type :file, content_type: VALID_IMAGE, if: :image?

  def paperclip_styles
    case typology
    when 'profile'
      return CUSTOM_SETTINGS[:USER_ATTACHMENT_IMAGE_SIZES]
    when 'cover'
      return CUSTOM_SETTINGS[:USER_ATTACHMENT_IMAGE_SIZES]
    when 'post_image'
      return CUSTOM_SETTINGS[:USER_ATTACHMENT_IMAGE_SIZES]
    end

    {}
  end

  # ...
end
```

**Comment by sidraval** [comment]
Thank you for reporting this. Unfortunately, we will be deprecating Paperclip and therefore will not have the bandwidth to address this issue.
